src/web/lib/braze/checkBrazeDependencies.ts

```diff
@@ -0,0 +1,100 @@
+import { getBrazeUuid } from '@root/src/web/lib/getBrazeUuid';
+import { hasRequiredConsents } from './hasRequiredConsents';
+import { hideSupportMessaging } from './hideSupportMessaging';
+
+type SuccessResult = {
+	isSuccessful: true;
+	data: ResultData;
+};
+
+type FailureResult = {
+	isSuccessful: false;
+	failureField: string;
+	failureData: any;
+	data: ResultData;
+};
+
+type DependenciesResult = SuccessResult | FailureResult;
+
+type ResultData = { [key: string]: string | boolean };
+
+type DependencyResult = {
+	name: string;
+	value: Promise<string | boolean | null | undefined>;
+};
+const buildFailureResponse = (name: string, value: any, data: ResultData) => ({
+	isSuccessful: false,
+	failureField: name,
+	failureData: value,
+	data,
+});
```

Comment:
personal preferrence but maybe the response could be structured like:

```
{
    isSuccessful: false,
    error: {
         failureField: name,
	failureData: value,
    },
    data
}
```

Reply:
that way if there are no errors we could have something that looks like:

```
{
    isSuccessful: true,
    error: null,
    data
}
```

Its nicer to not have to check if an object key exists and always assume it does exist  but is not set

Reply:
Oh interesting, yeah combining the two failure fields together is nice. Maybe the types as they are right now are overdoing it a bit. Currently I've got:

```js
type SuccessResult = {
	isSuccessful: true;
	data: ResultData;
};

type FailureResult = {
	isSuccessful: false;
	failureField: string;
	failureData: any;
	data: ResultData;
};

type DependenciesResult = SuccessResult | FailureResult;
```

But given a single nullable error key this could become:

```js
type DependencyError = {
        failureField: string;
	failureData: any;
};

type DependenciesResult = {
        error: DependencyError | null;
	data: ResultData;
}
```

The consumer would then just check if `error` is truthy to know the overall outcome.

> Its nicer to not have to check if an object key exists and always assume it does exist but is not set

I think in this case TypeScript is helping us, since if `isSuccessful` is `false` the type of `FailureResult` specifies that the `failureField` and `failureData` fields will always be defined. (Interesting blog post describing the approach I used here: https://mariusschulz.com/blog/more-literal-types-in-typescript#boolean-literal-types)

Reply:
I went ahead and combined the two failure fields into one object which I agree is nicer, thanks for the suggestion! For now I've kept separate `SuccessResult` and `FailureResult` types.